lhotse/cut.py

```diff
@@ -0,0 +1,218 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Iterable, Any, Union
+from uuid import uuid4
+
+import numpy as np
+import yaml
+
+from lhotse.features import Features, FeatureSet, overlay_fbank, pad_shorter
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Seconds, Decibels, overlaps, TimeSpan, overspans, Pathlike, asdict_nonull
+
+
+# One of the design principles for Cuts is a maximally "lazy" implementation, e.g. when overlaying/mixing Cuts,
+# we'd rather sum the feature matrices only after somebody actually calls "load_features".
+
+@dataclass
+class Cut:
+    """
+    A Cut is a single segment that we'll train on. It contains the features corresponding to
+    a piece of a recording, with zero or more SupervisionSegments.
+    """
+    id: str
+    channel: int
+
+    # Begin and duration are needed to specify which chunk of features to load.
+    start: Seconds
+    duration: Seconds
+
+    # The features can span longer than the actual cut - the Features object "knows" its start and end time
+    # within the underlying recording. We can expect the interval [begin, begin + duration] to be a subset of the
+    # interval represented in features.
+    features: Features
+
+    # Supervisions that will be used as targets for model training later on. They don't have to cover the whole
+    # cut duration. They also might overlap.
+    supervisions: List[SupervisionSegment]
+
+    def end(self) -> Seconds:
+        return self.start + self.duration
+
+    def load_features(self, root_dir: Optional[Pathlike] = None) -> np.ndarray:
+        return self.features.load(root_dir=root_dir, start=self.start, duration=self.duration)
+
+    def truncate(
+            self,
+            *,
+            offset: Seconds = 0.0,
+            until: Optional[Seconds] = None,
+            keep_excessive_supervisions: bool = True
+    ) -> 'Cut':
+        new_start = self.start + offset
```

Comment:
I'm not sure how I'd find documentation?

Reply:
.. and I don't understand the '*'

Reply:
Hmm yeah in some places I might've forgotten about the docs. Let me revise the PR before you go further, I'll add relevant docstrings. 

The '*' was introduced in Python 3 and means keyword-only arguments are allowed after it; i.e. f(10) won't work, you have to use f(until=10). The reason why I wanted to use that is that there are multiple arguments of the same type and it's easy to mix them up accidentally. As you can see I generally have a strong preference for using keyword args to avoid mixups (especially when refactoring) and increase readability.